AppController/djinn.rb

```diff
@@ -4883,7 +4890,11 @@ def check_stopped_apps()
       if my_node.is_load_balancer?
         stop_xmpp_for_app(app)
         Nginx.remove_app(app)
+
+        # Since the removal of an app from HAProxy can cause a reset of
+        # the drain flags, let's set them again.
         HAProxy.remove_app(app)
+        reset_drain
```

Comment:
Nope. The LBs tries to stop requests from going to the AppServers, but eventually (when the timeout expires) the AppEngine node will decide to yank the AppServer no matter what. So if a requests slips through, it will get interrupted.